### ISSUE TITLE
Music is playing but there is no audio

### ISSUE THREAD

**Issue by MagentaPuppy** [UnicornsOnLSD/finamp#177]
iphone 8, ios 13

**Comment by Maxplosion** [comment]
logs from finamp? 
does it work in on other clients and if yes which?
jellyfin server logs? 
are you transcoding?
see a pattern here...?

**Comment by MagentaPuppy** [comment]
**Finamp logs:** https://paste.gg/p/anonymous/c238f1a4142447909d5abae177e9166e
**Finamp audio logs:** https://paste.gg/p/anonymous/4ab3e8c877bb41bdbcdb6dd737de7c0e
**Jellyfin logs:** https://paste.gg/p/anonymous/3c221e7311fe4066a5346fe36e81b27a

It works on Jellyfin web client and ios client, i don't have and android phone to try Finamp on

not transcoding, audio plays if i turn it on

**Comment by UnicornsOnLSD** [comment]
What codec is your audio? OPUS and Vorbis aren't supported yet on iOS as iOS doesn't natively support those formats (it's dumb I know)

**Comment by MikePadge** [comment]
@UnicornsOnLSD thanks for your work on this app, I love it

I'm not super familiar with Flutter, but this might be a solution
https://github.com/EPNW/opus_flutter/tree/master/opus_flutter

**Comment by UnicornsOnLSD** [comment]
That's a really interesting looking package, I'll do some testing with it :)

**Comment by MagentaPuppy** [comment]
> What codec is your audio? OPUS and Vorbis aren't supported yet on iOS as iOS doesn't natively support those formats (it's dumb I know)

Not even transcoding it?? Also I’m kinda having this issue again, but now it only shows songs I’ve hearted on jellyfin web/iOS app, and yeah nothing opus is playing, annoying given that the vast majority of my music is opus

**Comment by UnicornsOnLSD** [comment]
Transcoding should work since Jellyfin will then serve it as AAC, which is supported. As I said, this is an iOS issue so it isn't likely to get fixed in the future. I believe the libopus package that was sent here was for when you want to encode Opus audio in your app, so it won't be useful here.

As for only showing favourite content, you can disable the favourite view by clicking on the star in the top right.

**Comment by UnicornsOnLSD** [comment]
Just noting here a comment I left on a duplicate issue - https://github.com/UnicornsOnLSD/finamp/issues/236#issuecomment-1174953175

As said there, it would be interesting to see if it would be possible to use [just_audio_mpv](https://github.com/bleonard252/just_audio_mpv). In theory it should be possible, but there has been a lot of discussion in just_audio regarding whether or not bundling MPV in apps is legal. I'd assume that the MPV guys aren't too bothered based on [mpv-android](https://play.google.com/store/apps/details?id=is.xyz.mpv)

**Comment by proline** [comment]
Apparently you can implement this by yourself I don't know it is complicated.
http://iosdeveloperzone.com/2012/08/07/precompiled-ogg-vorbis-libraries-for-ios/
The post is a little bit old. It could be a solution but if I understand you need to implement your own player.

**Comment by UnicornsOnLSD** [comment]
Yeah the libraries would only let me encode/decode those files directly, you can't plug them into Apple's player.

**Comment by fuomag9** [comment]
Maybe https://github.com/leinelissen/jellyfin-audio-player can be used? It's MIT and they have implemented opus playback

**Comment by jmshrv** [comment]
On iOS? It looks like they have the same issue - https://github.com/leinelissen/jellyfin-audio-player/issues/97

I've looked further into MPV, it'll probably be possible as it's LGPL2 (some bits are GPL only but we can safely ignore them). The concerns around legality come from GPL3's Tivoization-prevention stuff.

**Comment by fuomag9** [comment]
> On iOS? It looks like they have the same issue - [leinelissen/jellyfin-audio-player#97](https://github.com/leinelissen/jellyfin-audio-player/issues/97)
> 
> I've looked further into MPV, it'll probably be possible as it's LGPL2 (some bits are GPL only but we can safely ignore them). The concerns around legality come from GPL3's Tivoization-prevention stuff.

In my case I can successfully play opus files (maybe it's transcoding?)
With finamp I couldn't even with transcoding enabled in the app

**Comment by jmshrv** [comment]
> With finamp I couldn't even with transcoding enabled in the app

You're probably running into #269, which was fixed a while ago (but causes more issues in the form of #414 lol)

**Comment by fuomag9** [comment]
> > With finamp I couldn't even with transcoding enabled in the app
> 
> You're probably running into #269, which was fixed a while ago (but causes more issues in the form of #414 lol). It's not in a release yet thankfully.

Is that fixed in the AppStore version? I tried again but still couldn’t play music :(

**Comment by jmshrv** [comment]
It's not out yet, hopefully it will be soon. As I said, it breaks direct play at the moment so its probably a good thing that I was slow to publish it lol

**Comment by Chaphasilor** [comment]
@fuomag9 it should be out now. You can also try out the beta version which has support for transcoded downloads! :)

**Comment by gummykage** [comment]
Is it normal for OGG files to crackle and skip a bit?

**Comment by Chaphasilor** [comment]
I honestly don't know, but probably not. I guess this is on iOS? Does it also happen if you download them as a transcoded version?

**Comment by felix920506** [comment]
@gummykage I noticed the same issue and reported in #600

**Comment by willisplummer** [comment]
I'm running into this issue as well on ios

**Comment by felix920506** [comment]
Idk if anyone has tried this, but I got finamp to direct play opus on macos (installed the ios app via the app store)
I am on macos14.5 running finamp beta from testflight.

Steps are as follows:
1. If you already have the audio in opus codec, remux them to mp4 containers:
`ffmpeg -i something.opus -vn -c:a copy something.mp4`
Or webm container: `ffmpeg -i something.webm -vn -c:a copy something.mp4`

Alternatively, you can transcode other codecs to opus if desired, not recommended if source is lossy:
`ffmpeg -i something.flac -vn -c:a libopus something.mp4`

2. Rename the file to have .m4a extension. For example, `something.mp4` -> `something.m4a`. Straight rename is enough, no need to remux/transcode.

3. Place the resulting .m4a files on Jellyfin and rescan
4. Profit

**Comment by felix920506** [comment]
@sortedcord It is not direct playing Vorbis. The server is transcoding it to something else.

**Comment by Chaphasilor** [comment]
@felix920506 only just saw your comment, so essentially iOS/macOS can play OPUS as long as it's in a m4a container?

**Comment by felix920506** [comment]
@Chaphasilor As long as it's in mp4/m4a and stereo/mono above I think 17.4(not sure) and .caf on some older versions

**Comment by Chaphasilor** [comment]
Neat. I guess then we could use some sort of "direct streaming", letting the server repackage to m4a on the fly? Telling everyone to manually repackage their entire library doesn't seem like a great solution, and audiophiles tend to hate transcoding...

**Comment by willisplummer** [comment]
Using the beta, I enabled transcoding (Settings > Transcoding > Enable Transcoding). I also enabled it for downloads. Everything plays for me (though occasionally there is a long lag before a song starts - presumably because the server is underpowered). I don't think this issue needs to remain open, but I do think that the error messaging when a song can't be played because transcoding is disabled and it's an unsupported format could be improved. In the beta, I got an error but I think it was the generic "Could not play this track" or something to that effect

**Comment by Chaphasilor** [comment]
@willisplummer yeah, it is possible to play (and download) almost everything through transcoding. Before closing this issue I'd like to offer some sort of automatic transcoding, that transcodes (or repackages, if sufficient) the audio automatically if its in an unsupported format (or has a high bitrate). That feature is long overdue.

**Comment by felix920506** [comment]
@Chaphasilor by that point we are probably better of switching to the UniversalAudio endpoint on the server side since it automatically handles whether to direct play, remux or transcode when provided a list of supported codecs and containers.

**Comment by sortedcord** [comment]
@felix920506 do you have any suggestions as to what I should do? My library is predominantly OGG vorbis, if there is I can do anything apart from just re-encoding them to aac then that would be a life saver...

**Comment by Chaphasilor** [comment]
@felix920506 yeah that's definitely one valid approach. If there aren't any unforseen issues with that endpoint I see no reason why we wouldn't support it. We just haven't implemented it yet.

**Comment by felix920506** [comment]
@sortedcord turn on transcoding on your phone and forget about it. Let the server transcode to aac on the fly when you play.

**Comment by sortedcord** [comment]
@felix920506  I have a few playlists that contain both ogg vorbis and flac files. When I download them (without transcoding) the ogg tracks also get downloaded so some of the tracks are not playable anymore and I don't want to transcode my flacs. Is it possible to make it so that it forces transcoding for ogg tracks when they are downloaded?

**Comment by felix920506** [comment]
@sortedcord i think downloads are always transcodes

**Comment by sortedcord** [comment]
> @sortedcord i think downloads are always transcodes 

Apparently not, we have the option to choose to download the original files or the transcoded aac files.

**Comment by sortedcord** [comment]
I suppose that could work out, thanks!

**Comment by jmshrv** [comment]
iOS 18.4 has added support for OPUS/Vorbis in OGG containers! The only formats that Finamp doesn't support now are ALAC on Android (since Android doesn't implement support), and the weird formats like DSDs

**Comment by Chaphasilor** [comment]
Combined with the new transcoding options (including high quality and high compatibility FLAC) and the native support for Vorbis and OPUS I'm going to close this issue now.  
If there are further problems we should create specific new issues for those.
